webpack/assets/javascripts/react_app/components/HostsIndex/BulkActions/disassociate/BulkDisassociateModal.js

```diff
@@ -0,0 +1,195 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import {
+  Modal,
+  Alert,
+  Button,
+  TextContent,
+  Text,
+  TreeView,
+} from '@patternfly/react-core';
+import { addToast } from '../../../ToastsList/slice';
+import { foremanUrl } from '../../../../common/helpers';
+import { translate as __ } from '../../../../common/I18n';
+import { BULK_DISASSOCIATE_KEY, bulkDisassociate } from './actions';
+import { APIActions } from '../../../../redux/API';
+import {
+  HOSTS_API_PATH,
+  API_REQUEST_KEY,
+} from '../../../../routes/Hosts/constants';
+import { failedHostsToastParams } from '../helpers';
+
+const BulkDisassociateModal = ({
+  isOpen,
+  closeModal,
+  selectAllHostsMode,
+  selectedCount,
+  selectedResults,
+  fetchBulkParams,
+}) => {
+  const dispatch = useDispatch();
+  const virtualHosts = selectedResults?.filter(
+    h => h.compute_resource_id && h.uuid
+  );
+  const physicalHosts = selectedResults?.filter(h => !virtualHosts.includes(h));
+  const selectedResultsEmpty = selectedResults?.length === 0;
+
+  const selectedTreeViewData = [
+    {
+      name: __('Selected hosts'),
+      id: 'selected-hosts-tree-view-title',
+      customBadgeContent: selectAllHostsMode ? 'All' : selectedCount,
+    },
+  ];
+  const applicableTreeViewData = [
+    {
+      name: __('Applicable hosts'),
```

Comment:
I think Applicable could be confusing. Perhaps write out what it actually is?
```suggestion
      name: __('Hosts associated to compute resources'),
```